parsl/executors/taskvine/manager.py

```diff
@@ -0,0 +1,468 @@
+            # Obtain task from ready_task_queue
+            try:
+                task = ready_task_queue.get(timeout=1)
+                logger.debug("Removing executor task from queue")
+            except queue.Empty:
+                logger.debug("Queue is empty")
+                continue
+            if task.exec_mode == 'regular':
+                # Create command string
+                launch_cmd = "python3 exec_parsl_function.py {mapping} {function} {argument} {result}"
+                if manager_config.init_command != '':
+                    launch_cmd = "{init_cmd} " + launch_cmd
+                command_str = launch_cmd.format(init_cmd=manager_config.init_command,
+                                                mapping=os.path.basename(task.map_file),
+                                                function=os.path.basename(task.function_file),
+                                                argument=os.path.basename(task.argument_file),
+                                                result=os.path.basename(task.result_file))
+                logger.debug("Sending executor task {} (mode: regular) with command: {}".format(task.executor_id, command_str))
+                try:
+                    t = Task(command_str)
+                except Exception as e:
+                    logger.error("Unable to create executor task (mode:regular): {}".format(e))
+                    finished_task_queue.put_nowait(VineTaskToParsl(executor_id=task.executor_id,
+                                                                   result_received=False,
+                                                                   result=None,
+                                                                   reason="task could not be created by taskvine",
+                                                                   status=-1))
+                    continue
+            elif task.exec_mode == 'serverless':
+                if not lib_installed:
+                    # Declare and install common library for serverless tasks.
+                    # Library requires an environment setup properly, which is
+                    # different from setup of regular tasks.
+                    # If shared_fs is True, then no environment preparation is done.
+                    # Only the core serverless code is created.
+                    poncho_env_path = _prepare_environment_serverless(manager_config, env_cache_dir, poncho_create_script)
+
+                    # Don't automatically add environment so manager can declare and cache the vine file associated with the environment file
+                    add_env = False
+                    serverless_lib = m.create_library_from_functions('common-parsl-taskvine-lib',
+                                                                     run_parsl_function,
+                                                                     poncho_env=poncho_env_path,
```

Comment:
I get an error here:

```
  File "/home/benc/parsl/src/parsl/parsl/executors/taskvine/manager.py", line 245, in _taskvine_submit_wait
    serverless_lib = m.create_library_from_functions('common-parsl-taskvine-lib',
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Manager.create_library_from_functions() got an unexpected keyword argument 'poncho_env'
```

I think that's because this parameter was recently introduced (https://github.com/cooperative-computing-lab/cctools/pull/3427 18 hours ago) and so this mode will only work with a future release of ndcctools?

Maybe it would be useful to document that somewhere as a minimum version requirement.